azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/message.py

```diff
@@ -0,0 +1,53 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class Message(object):
+    def __init__(
+        self, data, creation_time_utc, message_id=None, content_encoding=None, content_type=None
+    ):
+        """
+        :param data: The  data that constitutes the payload
+        :param creation_time_utc: Date and time the message was created
+        :param message_id: A user-settable identifier for the message used for request-reply patterns.Format: A case-sensitive string (up to 128 characters long) of ASCII 7-bit alphanumeric characters + {'-', ':', '.', '+', '%', '_', '#', '*', '?', '!', '(', ')', ',', '=', '@', ';', '$', '''}.
```

Comment:
>message_id [](start = 15, length = 10)

do we need to add details on what gets used if nothing is provided?  I assume a message_id gets auto-generated, but what about encoding and type?  Can those be inferred from data?  e.g. if data is a UTF-8 string, can we just assume type and encoding without the user having to tell us?

Reply:
that's debatable, in the spirit of saving bandwidth we should only set properties that the customer wants (applies to creation-time if it's not mandatory too). 

---
In reply to: [237661641](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r237661641) [](ancestors = 237661641)

Reply:
message_id : from documentation it looks like it is user settable. So it should be possible to auto generate as well as user set from before.

Mostly the content type and encoding is always explicitly stated. Need to discuss more on this.

---
In reply to: [237661641](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r237661641) [](ancestors = 237661641)

Reply:
message id has been changed to `self.message_id = message_id if message_id else uuid.uuid4()`

---
In reply to: [237723135](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r237723135) [](ancestors = 237723135,237661641)

Reply:
why would we populate it ourselves? it's just more bandwidth. if not provided, leave blank. 

---
In reply to: [238907119](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r238907119) [](ancestors = 238907119,237723135,237661641)

Reply:
i thought if not provided we should auto generate the id. But I am seeing no code in Node that does that. In C# there is a code but it is never called as far as I can see. I am going to revert this change to before.

---
In reply to: [238907590](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r238907590) [](ancestors = 238907590,238907119,237723135,237661641)

Reply:
thanks. we should *NOT* autogenerate it

---
In reply to: [239226953](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r239226953) [](ancestors = 239226953,238907590,238907119,237723135,237661641)